timestream.lua

```diff
@@ -0,0 +1,339 @@
+                simulating_desired_fps = true
+                prev_frames = df.global.world.frame_counter
+                prev_time = df.global.enabler.clock
+                rate = 1
+                if simulating_units == 1 or simulating_units == 2 then
+                    print("Unit simulation is on.")
+                    if simulating_units ~= 2 then
+                        df.global.debug_turbospeed = false
+                    end
+                end
+            end
+            ticks_left = rate - 1
+
+            eventNow = false
+            seasonNow = false
+            timestream = 0
+            if df.global.cur_season_tick < 3360 then
+                month = 1
+            elseif df.global.cur_season_tick < 6720 then
+                month = 2
+            else
+                month = 3
+            end
+            if loaded ~= true then
+                dfhack.timeout(1,"frames",function() update() end)
+                loaded = true
+            end
+        else
+            print('Time set to normal speed.')
+            loaded = false
+            df.global.debug_turbospeed = false
+        end
+            if debug_mode then
+            print("Debug mode is on.")
+        end
+    elseif code==SC_MAP_UNLOADED then
+    end
+end
+
+function update()
+    prev_tick = df.global.cur_year_tick
+    if rate ~= 1 or simulating_desired_fps then
+        if last_frame + 1 == df.global.world.frame_counter then
+            timestream = 0
+
+            --[[if rate < 1 then
+                if df.global.cur_year_tick - math.floor(df.global.cur_year_tick/10)*10 == 5 then
+                    if counter > 1 then
+                        counter = counter - 1
+                        timestream = -1
+                    else
+                        counter = counter + math.floor(ticks_left)
+                    end
+                end
+            else
+            --]]
+            --counter = counter + rate-1
+            counter = counter + math.floor(ticks_left)
+            while counter >= 10 do
+                counter = counter - 10
+                timestream = timestream + 1
+            end
+            --end
+            eventFound = false
+            for i=0,#df.global.timed_events-1,1 do
+                event=df.global.timed_events[i]
+                if event.season == df.global.cur_season and event.season_ticks <= df.global.cur_season_tick then
+                    if eventNow == false then
+                        --df.global.cur_season_tick=event.season_ticks
+                        event.season_ticks = df.global.cur_season_tick
+                        eventNow = true
+                    end
+                    eventFound = true
+                end
+            end
+            if eventFound == false then eventNow = false end
+
+            if df.global.cur_season_tick >= 3359 and df.global.cur_season_tick < 6719 and month == 1 then
```

Comment:
Should this check for `>= 3360` instead? I ask because I'm seeing an assignment to 3360 below. (Same goes for the other checks).

Regarding constants, maybe something like `SEASON_LENGTH` could work, and could be reused as `SEASON_LENGTH - 1`, `SEASON_LENGTH * 2`, etc.

Reply:
Was what I thought too at first, but then I figured he had some reason, so I didn't want to change something I didn't fully understand the consequences of.

Reply:
@IndigoFenix is fairly active on the forums (not sure about here), so you could ask there if you have an account. It's an old script, though.